store/store.go

```diff
@@ -531,11 +531,57 @@ func (s *SnapUbuntuStoreRepository) FindSnaps(searchTerm string, channel string,
 	return snaps, nil
 }
 
-// Updates returns the available updates for a list of snap identified by fullname with channel.
-func (s *SnapUbuntuStoreRepository) Updates(installed []string, auther Authenticator) (snaps []*snap.Info, err error) {
-	// XXX: uses obsolete end point!
+// RefreshCandidate contains information for the store about the currently
+// installed snap so that the store can decide what update we should see
+type RefreshCandidate struct {
+	SnapID   string
+	Revision snap.Revision
+	Epoch    string
+	DevMode  bool
 
-	jsonData, err := json.Marshal(map[string][]string{"name": installed})
+	// the desired channel
+	Channel string
+}
+
+// the exact bits that we need to send to the store
+type currentSnapJson struct {
+	SnapID   string `json:"snap_id"`
+	Channel  string `json:"channel"`
+	Revision int    `json:"revision,omitempty"`
+	Epoch    string `json:"epoch"`
+	DevMode  bool   `json:"devmode"`
```

Comment:
@niemeyer Yeah, I understand that snapd internally uses confinement and devmode this way, but somehow I feel this change makes the store need to be aware of that too. The store doesn't know anything about a devmode, but it sure should be able to understand confinements and decide which the right update (if any) for a given a snap setup description (ie. channel, confinement, epoch) is. The client could decide whatever value to pass for each.
AFAICT, the store could just return the expected response for the requested scenario (without knowing anything else but the data that it already stores and indexes from the snaps metadata).
That's what keeps me doubting from changing the original decided API. Makes sense?